src/promptflow-tracing/promptflow/tracing/_trace.py

```diff
@@ -104,14 +104,25 @@ def enrich_span_with_trace(span, trace: Trace):
 
 def enrich_span_with_prompt_info(span, func, kwargs):
     try:
-        # Assume there is only one prompt template parameter in the function,
-        # we use the first one by default if there are multiple.
-        prompt_tpl_param_name = get_prompt_param_name_from_func(func)
-        if prompt_tpl_param_name is not None:
-            prompt_tpl = kwargs.get(prompt_tpl_param_name)
-            prompt_vars = {
-                name: kwargs.get(name) for name in get_input_names_for_prompt_template(prompt_tpl) if name in kwargs
-            }
+        prompt_tpl, prompt_vars = None, None
+        if not inspect.isfunction(func):
+            if type(func).__name__ == "Prompty":
+                # Get prompt template from prompty
+                prompt_tpl = getattr(func, "_template")
+                prompt_vars = {
+                    name: kwargs.get(name) for name in get_input_names_for_prompt_template(prompt_tpl) if name in kwargs
+                }
```

Comment:
what could be in the missing else branch here？

Reply:
fix it, it the func is not prompty, it will use origin logic to get template and inputs